IdentityCore/src/webview/operations/MSIDWebResponseBrokerInstallOperation.m

```diff
@@ -0,0 +1,116 @@
+#import "MSIDDefaultTokenResponseValidator.h"
+#import "MSIDAccountMetadataCacheAccessor.h"
+
+#if TARGET_OS_IPHONE
+    #import "MSIDAppExtensionUtil.h"
+#endif
+
+@interface MSIDWebResponseBrokerInstallOperation()
+
+@property (nonatomic) NSURL *appInstallLink;
+
+@end
+
+@implementation MSIDWebResponseBrokerInstallOperation
+
+- (nullable instancetype)initWithResponse:(MSIDWebviewResponse *)response
+                                    error:(NSError **)error
+{
+    #if TARGET_OS_IPHONE
+        self = [super initWithResponse:response
+                                 error:error];
+        if (self)
+        {
+            if (![response isKindOfClass:MSIDWebWPJResponse.class] || [NSString msidIsStringNilOrBlank:[(MSIDWebWPJResponse *)response appInstallLink]])
+            {
+                return nil;
+            }
+            
+            MSIDWebWPJResponse *wpjResponse = (MSIDWebWPJResponse *)response;
+            _appInstallLink = [NSURL URLWithString:wpjResponse.appInstallLink];
+        }
+        
+        return self;
+    #else
+        return nil;
+    #endif
+}
+
+- (void)invokeWithInteractiveTokenRequestParameters:(MSIDInteractiveRequestParameters *)interactiveTokenRequestParameters
+                                         completion:(MSIDRequestCompletionBlock)completion
+{
+    if (!completion)
+    {
+        return;
+    }
+    
+    #if TARGET_OS_IPHONE
+        if ([interactiveTokenRequestParameters isKindOfClass:MSIDInteractiveTokenRequestParameters.class])
+        {
+            NSError *brokerError;
+            MSIDBrokerInteractiveController *brokerController = [[MSIDBrokerInteractiveController alloc] initWithInteractiveRequestParameters:(MSIDInteractiveTokenRequestParameters *)interactiveTokenRequestParameters
+                                                                                                                         tokenRequestProvider:[self createTokenRequestProdiving]
+                                                                                                                            brokerInstallLink:self.appInstallLink
+                                                                                                                                        error:&brokerError];
+            [brokerController acquireToken:completion];
+        } else
+        {
+            NSError *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, @"Wrong type of interactive request parameter", nil, nil, nil, nil, nil, YES);
+            completion(nil, error);
+        }
+    #else
+        NSError *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, @"Trying to install broker on macOS, where it's not currently supported", nil, nil, nil, nil, nil, YES);
+        completion(nil, error);
+    #endif
+}
+
+- (MSIDDefaultTokenRequestProvider *)createTokenRequestProdiving
+{
+    MSIDKeychainTokenCache *dataSource = MSIDKeychainTokenCache.defaultKeychainCache;
```

Comment:
This won't be default cache all the time. It needs to use right group.

Reply:
I was reading the ```MSAIStorageManagerImpl``` in CPP and on non-mac target, they simply create a ```dataSource``` by calling ```[MSAIMSIDKeychainTokenCache new]```, which use the default adal group. 

Do you mean there is a function I can get the right group somehow? The default group is the only one I saw in the cpp code base. Might misunderstood some import parts, please guide.

Reply:
Only MSAL will know the right group (MSAL Obj-C and MSAL CPP), so default group shouldn't be used.

Reply:
See below comments